sdks/python/apache_beam/runners/interactive/extensions/apache-beam-jupyterlab-sidepanel/src/clusters/Clusters.tsx

```diff
@@ -0,0 +1,289 @@
+ * 1. View all Dataproc clusters managed by Interactive Beam
+ * 2. Delete a selected cluster. This will "reset" the cluster used by the
+ *    corresponding pipelines and the cluster will be recreated when the
+ *    pipeline is executed again.
+ * 3. Select a default cluster. This is the cluster that will be used by
+ *    Interactive Beam when no master_url or Google Cloud project is
+ *    specified in the pipeline options.
+ */
+export class Clusters extends React.Component<IClustersProps, IClustersState> {
+  constructor(props: IClustersProps) {
+    super(props);
+    this._inspectKernelCode =
+      'from apache_beam.runners.interactive ' +
+      'import interactive_environment as ie\n' +
+      'ie.current_env().inspector.list_clusters()';
+    this._model = new KernelModel(this.props.sessionContext);
+    this.state = {
+      kernelDisplayName: 'no kernel',
+      clusters: this._model.executeResult,
+      defaultClusterId: '',
+      selectedId: '',
+      selectedName: '',
+      showDialog: false,
+      displayTable: true
+    };
+  }
+
+  componentDidMount(): void {
+    this._queryKernelTimerId = setInterval(
+      () => this.queryKernel(this._inspectKernelCode),
+      2000
+    );
+    this._updateRenderTimerId = setInterval(() => this.updateRender(), 1000);
+    this._updateSessionInfoTimerId = setInterval(
+      () => this.updateSessionInfo(),
+      2000
+    );
+  }
+
+  componentWillUnmount(): void {
+    clearInterval(this._queryKernelTimerId);
+    clearInterval(this._updateRenderTimerId);
+    clearInterval(this._updateSessionInfoTimerId);
+  }
+
+  queryKernel(code: string): void {
+    this._model.execute(code);
+  }
+
+  updateRender(): void {
+    const clustersToUpdate = this._model.executeResult;
+    if (Object.keys(clustersToUpdate).length) {
+      this.setState({ displayTable: true });
+      if (
+        JSON.stringify(this.state.clusters) !== JSON.stringify(clustersToUpdate)
+      ) {
+        this.setState({ clusters: clustersToUpdate });
+      }
+    }
+  }
+
+  updateSessionInfo(): void {
+    if (this.props.sessionContext) {
+      const newKernelDisplayName = this.props.sessionContext.kernelDisplayName;
+      if (newKernelDisplayName !== this.state.kernelDisplayName) {
+        this.setState({
+          kernelDisplayName: newKernelDisplayName
+        });
+      }
+    }
+  }
+
+  setDefaultCluster(cluster_id: string): void {
+    const setDefaultClusterCode =
+      'ie.current_env().clusters.set_default_cluster' + `('${cluster_id}')`;
+    this.queryKernel(setDefaultClusterCode);
+    this.setState({ defaultClusterId: cluster_id });
+  }
+
+  displayDialog(open: boolean, key: string, clusterName: string): void {
+    this.setState({
+      showDialog: open,
+      selectedId: key,
+      selectedName: clusterName
+    });
+  }
+
+  deleteCluster(cluster_id: string): void {
+    const deleteClusterCode =
+      'ie.current_env().clusters.delete_cluster' + `('${cluster_id}')`;
```

Comment:
Got it! This would be better. I've now changed the inspector method to get_cluster_master_url() as it is easier to use the ib.clusters mappings with a master_url.